api/src/main/java/org/openmrs/module/kenyaemr/reporting/data/converter/definition/evaluator/fp/FPTBScreeningEvaluator.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.kenyaemr.reporting.data.converter.definition.evaluator.fp;
+
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.kenyaemr.reporting.data.converter.definition.fp.FPTBScreeningDefinition;
+import org.openmrs.module.reporting.data.encounter.EvaluatedEncounterData;
+import org.openmrs.module.reporting.data.encounter.definition.EncounterDataDefinition;
+import org.openmrs.module.reporting.data.encounter.evaluator.EncounterDataEvaluator;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.EvaluationException;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+import java.util.Map;
+
+@Component
+@Handler(supports= FPTBScreeningDefinition.class, order=50)
+public class FPTBScreeningEvaluator implements EncounterDataEvaluator {
+    @Autowired
+    private EvaluationService evaluationService;
+
+    @Override
+    public EvaluatedEncounterData evaluate(EncounterDataDefinition encounterDataDefinition, EvaluationContext context) throws EvaluationException {
+        EvaluatedEncounterData c = new EvaluatedEncounterData(encounterDataDefinition, context);
+        String qry = "SELECT f.encounter_id, c.name AS resulting_tb_status " +
+                "FROM kenyaemr_etl.etl_family_planning f " +
+                "JOIN kenyaemr_etl.etl_tb_screening t ON f.patient_id = t.patient_id " +
+                "JOIN openmrs.concept_name c ON t.resulting_tb_status = c.concept_id " +
```

Comment:
Transforn concept  id to name directly. The join may not be accurate. I think you need to relate FP visit to a specific row in TB screening, most likely, screening that happened within the same period or day